code/modules/antagonists/traitor/components/demoraliser.dm

```diff
@@ -0,0 +1,151 @@
+/*
+ * Applies a role-based mood if you can see the parent.
+ *
+ * - Applies a mood to people who are in visible range of the item.
+ * - Does not re-apply mood to people who already have it.
+ * - Sends a signal if a mood is successfully applied.
+ */
+/datum/proximity_monitor/advanced/demoraliser
+	var/datum/demoralise_moods/moods
+
+/datum/proximity_monitor/advanced/demoraliser/New(atom/_host, range, _ignore_if_not_on_turf = TRUE, datum/demoralise_moods/moods)
+	. = ..()
+	src.moods = moods
+	RegisterSignal(host, COMSIG_PARENT_EXAMINE, .proc/on_examine)
+
+/datum/proximity_monitor/advanced/demoraliser/Destroy()
+	UnregisterSignal(host, COMSIG_PARENT_EXAMINE)
+	return ..()
+
+/datum/proximity_monitor/advanced/demoraliser/field_turf_crossed(atom/movable/crossed, turf/location)
+	if (!ishuman(crossed))
+		return
+	if (!can_see(crossed, host, current_range))
+		return
+	on_seen(crossed)
+
+/*
+ * Signal proc for [COMSIG_PARENT_EXAMINE].
+ * Immediately tries to apply a mood to the examiner, ignoring the proximity check.
+ * If someone wants to make themselves sad through a camera that's their choice I guess.
+ */
+/datum/proximity_monitor/advanced/demoraliser/proc/on_examine(datum/source, mob/examiner)
+	SIGNAL_HANDLER
+	if (ishuman(examiner))
+		on_seen(examiner)
```

Comment:
Why do we limit this to humans? What if non-humans also get mood in the future? (Ex: xenos)